client/components/Profiles/ListedProfiles.tsx

```diff
@@ -0,0 +1,68 @@
+import { View, Text, FlatList, StyleSheet } from 'react-native'
+import React, { useEffect } from 'react'
+import { useDataContext } from '@/context/globalContext';
+import { getAllProfiles } from '@/services/apiService';
+import ProfileButton from './ProfileButton';
+import { ProfileType } from '@/types';
+
+
+const ListedProfiles = () => {
+    const { user, profiles, setProfiles, setSelectedProfile } = useDataContext();;
+
+    useEffect(()=> {
+        const getProfiles = async () => {
+            if (user) {
+              const profiles = await getAllProfiles(user.id);
+              console.log('user id connected:', user.id)
+              setProfiles(profiles);
+            }
+        }
+
+        getProfiles();
+    }, [user]);
+
+
+    const handleProfilePress = (profile: ProfileType) => {
+        setSelectedProfile(profile);
+        console.log('profile selected:', profile)
+      };
+
+    if (!user) {
+        return <Text>Loading user data...</Text>
+    }
+
+    if (!profiles || profiles.length === 0) {
+        return <Text>No profiles available, please create new profile.</Text>
```

Comment:
TODO: should redirect to create profile page

Reply:
If you refer to the profilesScreen.tsx, you'll see the create new profile button is rendered directly above this text